packages/ai/core/util/chat-store.ts

```diff
@@ -0,0 +1,407 @@
+import { generateId as generateIdFunction, parsePartialJson } from '.';
+import type {
+  ReasoningUIPart,
+  TextUIPart,
+  ToolInvocation,
+  ToolInvocationUIPart,
+  UIMessage,
+} from '../types';
+import { throttle } from './throttle';
+
+interface ChatStoreSubscriber {
+  onChatMessagesChanged(): void;
+  // onChatStateChanged?
+  // onActiveChatChanged?
+  // onChatAdded?
+  // onChatRemoved?
+}
+
+type ChatStoreEvent = 'chat-messages-changed';
+
+const ChatStoreEventMap = {
+  'chat-messages-changed': 'onChatMessagesChanged',
+} as const;
+
+interface ChatStoreInitialization {
+  chatId?: string;
+  initialMessages?: UIMessage[];
+  throttleMs?: number;
+}
+
+/**
+ * Internal class for managing UIMessages
+ */
+export class ChatStore {
+  private chatId: string;
+  private messages: UIMessage[];
+  private subscribers: Set<ChatStoreSubscriber>;
+  private notify: (event: ChatStoreEvent) => void;
```

Comment:
I was thinking that the chat store can have multiple chats (and maybe an `activeChat`)